### PR TITLE
floorp: init at 11.5.0

### LINKED ISSUE #255940: Package request: floorp
**Project description**
Floorp is a firefox fork.

**Metadata**

* homepage URL: https://floorp.app/en/
* source URL: https://github.com/floorp-Projects/floorp/
* license: MPLv2.0
* platforms: windows, macos, linux

### PR DESCRIPTION
**Pull request by christoph-heiss**
## Description of changes

Closes #255940.

This adds [Floorp](https://floorp.app/), which is another fork of Firefox.
It describes itself as being "built on Firefox, built in Japan and is a new browser with excellent privacy & flexibility".

Due to it having its own versioning system, a small adjustment for `buildMozillaMach` was needed - hopefully that is fine.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by MikaelFangel** [comment]
Shouldn't this use the new by-name hierarchy instead of the old? 😇

**Comment by MikaelFangel** [comment]
@christoph-heiss - Just asked in the matrix channel and it should use the old hierarchy for now. Sorry for the confusion.

**Comment by christoph-heiss** [comment]
@MikaelFangel No worries, I just pushed the original changes again.
Thanks for confirming via Matrix!

**Review by MikaelFangel — APPROVED** [review]

File: pkgs/applications/networking/browsers/firefox/common.nix [hunk 1]
```diff
@@ -1,5 +1,6 @@
 { pname
 , version
+, pversion ? version
```
Comment (mweinelt) [inline pkgs/applications/networking/browsers/firefox/common.nix:3]:
```suggestion
, packageVersion ? version
```

**Comment by mweinelt** [comment]
Note that submitting a browser, a very sensitive program, incurs a certain maintenance cost. Building Firefox from source takes roughly an hour even on a beefy machine. And floorp seems to do multiple releases per month. Ideally you'd do some manual testing with a fresh profile (likely `floorp -P`) when the firefox base gets updated. Also note that the Firefox nixos test sometimes fails due to exceeding the fdlimit of the environment.

When there are security issues they should be fixed in a timely manner (<7 days) within nixpkgs. When we fall behind on maintenance we leave our users exposed. We require that browser updates are backported to the stable release branch.

**Comment by christoph-heiss** [comment]
> Note that submitting a browser, a very sensitive program, incurs a certain maintenance cost. Building Firefox from source takes roughly an hour even on a beefy machine. And floorp seems to do multiple releases per month. 

Yeah, this shouldn't be _that_ much of a problem for me - I luckily got a beefy work station, a fresh build of floorp takes <30min here for me.

> When there are security issues they should be fixed in a timely manner (<7 days) within nixpkgs. When we fall behind on maintenance we leave our users exposed. We require that browser updates are backported to the stable release branch.

I understand. I've already read up on the backporting guide in `CONTRIBUTING.md` and looked at some PRs marked for backporting. 

I am also currently in contact with them if there are any (possibly limited to package maintainers & co) proper announcement channels, such as a mailing list. 
For now, I'm subscribed to the RSS feed of their [release announcement blog](https://blog.ablaze.one/category/ablaze/ablaze-project/floorp/) and watching GitHub any releases. 

Providing security fixes timely should then be of no problem - if the security fixes do get properly announced, that is, of course. But from a cursory glance at the history of release announcements, this seems to be the case. (E.g. the most recent one, 11.4.1; which was released solely because of CVE-2023-5217).

Happy to discuss this further as needed, I'm still mostly new to this aspect of package maintenance. 

Lastly, one issue that I still have to figure out is that `floorp --version` still reports Mozilla as vendor. 
```
$ ./result/bin/floorp --version
Mozilla Floorp 11.4.1
```

For above reason, I would leave this open for a bit, to also see if there is interest from other people too, maybe even signing up for co-maintaining. I'm committed, but one can also never know what life suddenly brings.

**Comment by Ultra980** [comment]
> Lastly, one issue that I still have to figure out is that `floorp --version` still reports Mozilla as vendor.
> 
> ```
> $ ./result/bin/floorp --version
> Mozilla Floorp 11.4.1
> ```

The official binary also reports this.

**Comment by christoph-heiss** [comment]
> The official binary also reports this.

Oh, well. I actually didn't cross-check the version with an official build - but that settles that.
Thanks!

**Comment by Henry-Hiles** [comment]
I can test this later this week, and add an approval if all goes well.

**Comment by christoph-heiss** [comment]
Updated to v11.5.0, which was released ~20h ago.

**Comment by Henry-Hiles** [comment]
I'm trying to test, but god this takes a long time to build...

**Comment by Henry-Hiles** [comment]
This is the correct command, right? `nix shell "github:christoph-heiss/nixpkgs?ref=new/floorp#floorp"`

**Review by Henry-Hiles — APPROVED** [review]
It finally built, LGTM!

**Comment by Henry-Hiles** [comment]
BTW are we waiting for something before a merge or is it just a matter of time?

**Comment by christoph-heiss** [comment]
> BTW are we waiting for something before a merge or is it just a matter of time?

From my side, all issues/questions are addressed and ready to go.

**Pull request merged**